include/boost/geometry/algorithms/detail/is_simple/linear.hpp

```diff
@@ -285,19 +285,17 @@ struct is_simple_multilinestring
     static inline bool apply(MultiLinestring const& multilinestring,
                              Strategy const& strategy)
     {
-        typedef per_linestring<Strategy> per_ls;
-
         // check each of the linestrings for simplicity
         // but do not compute self-intersections yet; these will be
         // computed for the entire multilinestring
-        if ( ! detail::check_iterator_range
-                 <
-                     per_ls, // do not compute self-intersections
-                     true // allow empty multilinestring
-                 >::apply(boost::begin(multilinestring),
-                          boost::end(multilinestring),
-                          per_ls(strategy))
-             )
+        // return true for empty multilinestring
+
+        using not_simple = not_simple<Strategy>; // do not compute self-intersections
```

Comment:
This would be `std::unary_negate<is_simple<Strategy>>` and it'd have to be created with: `std::unary_negate<is_simple<Strategy>>(is_simple<Strategy>(strategy))` making it more complex than the function call below.

Reply:
If we don't intend to use it anywhere else it doesn't have to be flexible. It makes no sense to add complexity only because we might some day in the future do something with it.

Reply:
Besides is_simple + negate to represent is_not_simple is as flexible as is_not_simple + negate to represent is_simple. It is just a matter of preference, not flexibility.

Reply:
It wouldn't be that complex if we used `std::not1`, i.e. `std::not1(is_simple<Strategy>(strategy))`, but this function is deprecated in C++17. In C++17 `std::not_fn` was introduced but we can't use that. So we'd be forced to write our own utility if we wanted this to look nice.